Code/Framework/AzCore/Tests/Time/TimeTests.cpp

```diff
@@ -77,4 +76,191 @@ namespace UnitTest
         int64_t delta = static_cast<int64_t>(timeMs) - static_cast<int64_t>(timeUsToMs);
         EXPECT_LT(abs(delta), 1);
     }
-}
+
+    class TimeSystemTests : public AllocatorsTestFixture
+    {
+    public:
+        void SetUp() override
+        {
+            SetupAllocator();
+            m_controlTime = static_cast<AZ::TimeUs>(AZStd::GetTimeNowMicroSecond());
+            m_timeSystem = AZStd::make_unique<AZ::TimeSystem>();
+        }
+
+        void TearDown() override
+        {
+            m_controlTime = AZ::Time::ZeroTimeUs;
+            m_timeSystem.reset();
+            TeardownAllocator();
+        }
+
+        AZ::TimeUs GetDiff(AZ::TimeUs time1, AZ::TimeUs time2)
+        {
+            // AZ::TimeUs is unsigned so make sure to not underflow.
+            return time1 > time2 ? time1 - time2 : time2 - time1;
+        }
+
+        AZ::TimeUs m_controlTime;
+        AZStd::unique_ptr<AZ::TimeSystem> m_timeSystem;
+    };
+    
+    TEST_F(TimeSystemTests, TestGetRealElapsedTimeUs)
+    {
+        // sleep for a bit to advance time.
+        AZStd::this_thread::sleep_for(AZStd::chrono::milliseconds(2));
+
+        // find the delta for the control and from GetRealElapsedTimeUs
+        const AZ::TimeUs baseline = static_cast<AZ::TimeUs>(AZStd::GetTimeNowMicroSecond()) - m_controlTime;
+        const AZ::TimeUs elapsedTime = m_timeSystem->GetRealElapsedTimeUs();
+
+        const AZ::TimeUs diff = GetDiff(baseline, elapsedTime);
+
+        // elapsedTime should be within 10 microseconds from baseline.
+        EXPECT_LT(diff, AZ::TimeUs{ 10 });
+    }
+
+    TEST_F(TimeSystemTests, TestGetElapsedTimeUs)
```

Comment:
uber nit: Including 'Test' in test names tends to be redundant and could just be deleted